daemon/daemon.go

```diff
@@ -479,6 +479,7 @@ func (daemon *Daemon) newContainer(name string, config *runconfig.Config, imgID
 	var (
 		id  string
 		err error
+		ane = name == ""
```

Comment:
At the function level, we are really checking if a name was not passed for the container. Container will never be anonymous. I should then probably choose `nameNotSpecified` or `noExplicitName`, `nonexplicitlyNamed` in this context. I do not like they are so long...